Sources/Navidux/Core/Navigation/NavigationStore/NavigationStore.swift

```diff
@@ -0,0 +1,16 @@
+//
+//  NavigationStore.swift
+//
+//
+//  Created by Петр Тартынских on 01.03.2024.
+//
+
+public struct NavigationStore {
+    public var isAlertShow: Bool = false
+    public var hasOverlay: Bool = false
+    
+    public init(isAlertShow: Bool = false, hasOverlay: Bool = false) {
+        self.isAlertShow = isAlertShow
+        self.hasOverlay = hasOverlay
+    }
+}
```

Comment:
Я думаю будет хорошей идеей подписать `Store` под `ObservableObject`, чтобы юзеры могли подписываться под обновления. Быстренько набросал то как я вижу эти изменения:

```suggestion
public final class NavigationStore: ObservableObject {
    @Published public var isAlertPresented: Bool = false
    @Published public var isModalPresented: Bool = false
}
```